packages/ingest/HttpProviderClient.js

```diff
@@ -42,7 +42,15 @@ class HttpProviderClient {
   list(path) {
     validateHost(this.host);
 
-    const pattern = /<a href="([^>]*)">[^<]+<\/a>/;
+    // Make pattern case-insensitive and return all matches
+    // instead of just first one
+    const matchLinksPattern = /<a href="([^>]*)">[^<]+<\/a>/ig;
+    const matchLeadingSlashesPattern = /^\/+/;
+
+    // Some providers provide files with one number after the dot (".")
+    // (e.g. tmtdayacz8110_5.6)
+    // TODO: Why are we assuming a file extension between 1 and 4 characters long?
```

Comment:
Does anyone know why we are making this assumption?

Reply:
No.  This seems arbitrary.     Particularly given the regexp correctly has the \s*$ pattern.... it's an odd place to explicitly stop the match.  It also implies target files *must* have one.

Reply:
Seems like the users are in favor of dropping this assumption entirely, which makes sense since discover-granules will only return files that match the `granuleIdExtraction` as discovered